bin/sirius-script-rabpm-mngr.py

```diff
@@ -129,34 +138,25 @@ def main():
                        help="Reset a batch of specified afc devices (timing, fofb or bpm)")
     add_rack_arg(parser_boot_from_flash)
 
-    output = subprocess.check_output(['ssh-agent'])
-    agent_env = parse_agent_env(output)
-    os.environ.update(agent_env)
-
-    # Use shell=True, so the tilde (~) is correctly expanded by having the
-    # shell interpret the command. The command must be a string instead of a
-    # list so it becomes a single argument to `/bin/sh -c`.
-    subprocess.run("ssh-add ~/.ssh/id_ed25519_rabpm", shell=True)
-
     args = parser.parse_args()
 
+    mngr = RABPMMngr()
+
     for rack in args.racks:
         if args.command == "afc-list":
-            run_pcie_list(rack)
+            mngr.run_pcie_list(rack)
         elif args.command == "afc-rescan":
-            run_pcie_rescan(rack)
+            mngr.run_pcie_rescan(rack)
         elif args.command == "afc-remove":
-            run_pcie_remove(rack, args.slot)
+            mngr.run_pcie_remove(rack, args.slot)
         elif args.command == "ioc-restart":
-            run_ioc_restart(rack, args.ioc, args.slot)
+            mngr.run_ioc_restart(rack, args.ioc, args.slot)
         elif args.command == "rffe-reset":
-            run_rffe_reset(rack, args.vslots)
+            mngr.run_rffe_reset(rack, args.vslots)
         elif args.command == "afc-reset":
-            run_boot_from_flash(rack, args.slots, args.device)
+            mngr.run_boot_from_flash(rack, args.slots, args.device)
 
-    # Point stdout/stderr to DEVNULL so the output is not printed
-    subprocess.run(["ssh-agent", "-k"],
-                   stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+    mngr.cleanup_ssh_agent()
```

Comment:
This works as the resource manager I mentioned. Just do `with RaBPMMngr() as mngr:`